File: 1218 (Longest Arithmetic Subsequence of Given Difference)/cpp/Solution.cpp
class Solution {
public:
    int longestSubsequence(vector<int>& arr, int difference) {
        int res= 1;
        unordered_map<int,int> dic;
        for(int i =0;i<arr.size();i++){
            int cnt = 1;
            if (dic.find(arr[i]-difference) != dic.end()) cnt = 1+dic[arr[i]-difference];
            dic[arr[i]] = cnt;
            res = max(res, cnt);
        }
        return res;
    }
};

File: 1218 (Longest Arithmetic Subsequence of Given Difference)/python/Solution.py
class Solution:
    def longestSubsequence(self, arr: List[int], difference: int) -> int:
        dic = {}
        res = 1

        for i in range(len(arr)):
            cnt = dic.get(arr[i]-difference, 0) +1
            dic[arr[i]] = cnt
            res= max(res, cnt)
        return res